### DISCUSSION TITLE
$Where filter using function not working in MongoDB

### DISCUSSION THREAD

**Discussion by Lishenga** [mikro-orm/mikro-orm#2879] [Q&A]
Hello am trying to search and filter using the $where filter in MikroOrm with MongoDB and it is not executing. Could you tell me where the error is with my code. Thank you 

```ts
let res: Cursor<any>;
res = this.em.getDriver().getConnection().getCollection(Person.name).find({
            $where: function () {
                var deepIterate = function (obj, value) {
                    for (var field in obj) {
                        if (String(obj[field]).includes(value)) {
                            return true;
                        }
                        var found = false;
                        if (typeof obj[field] === 'object') {
                            found = deepIterate(obj[field], value)
                            if (found) { return true; }
                        }
                    }
                    return true;
                };
                // return deepIterate(this, search)
            }
        }).skip(offset).limit(limit);

const items. = await res.toArray()
```

The above query doesn't work, what could be issue ? Thank you

**Comment by B4nan** [comment]
You are working with the native mongo db api, so this is not really the right place to ask why its not working. There is nothing involved fromt the ORM code, `getCollection()` will give you the native `Collection` instance from the driver directly.

I would guess the problem is you are not returning anything from the `$where` function, not even sure if its allowed to use callbacks there.
